packages/itwin/grouping-mapping-widget/src/widget/components/GroupAction.tsx

```diff
@@ -218,7 +220,69 @@ const GroupAction = ({
     mappingClient,
   ]);
 
-  const isBlockingActions = !(details.groupName && (query || simpleSelectionQuery) && !isRendering && !isLoading);
+  const queryGenerationComponent = () => {
+    switch (localQueryGenerationType) {
+      case "Selection": {
+        return (
+          <GroupQueryBuilderContext.Provider
+            value={{
+              currentPropertyList,
+              setCurrentPropertyList,
+              query,
+              setQuery,
+              queryBuilder,
+              setQueryBuilder,
+              isLoading,
+              isRendering,
+              resetView,
+            }}
+          >
+            <GroupQueryBuilderContainer />
+          </GroupQueryBuilderContext.Provider>
+        );
+      }
+      case "Search": {
+        return (
+          <SearchUIProvider
+            updateQuery={updateQuery}
+            isUpdating={isUpdating}
+            resetView={resetView}
+          />
+        );
+      }
+      case "Manual": {
+        return (
+          <ManualUIProvider
+            updateQuery={updateQuery}
+            isUpdating={isUpdating}
+            resetView={resetView}
+          />
+        );
+      }
+      default: {
+        if (localQueryGenerationType && localQueryGenerationType.length > 0) {
+          const selectedUIProvider = uiProviders.find(
+            (e) => e.name === localQueryGenerationType,
+          );
+          if (selectedUIProvider) {
+            return React.createElement(selectedUIProvider.uiComponent, {
+              updateQuery,
+              isUpdating: isLoading || isRendering,
```

Comment:
You already have a local variable called `isUpdating` which evaluates to the same value as this. Reuse it.

Reply:
good catch

Reply:
You missed this in `getRadioTileComponent`.